AndroidAssignment/app/src/main/java/org/sopt/ui/view/user/SignUpActivity.kt

```diff
@@ -14,14 +18,79 @@ class SignUpActivity : BaseActivity<ActivitySignUpBinding, UserViewModel>() {
     override val viewModel: UserViewModel by viewModels()
 
     override fun initView() {
+        initFocusEvent()
+        initClickEvent()
+    }
+
+    private fun initFocusEvent() {
+        binding.etName.setOnFocusChangeListener { _, hasFocus ->
+            if(hasFocus)
+                binding.clUserName.setBackgroundResource(R.drawable.rectangle_border_sky_10)
+            else {
+                if(isEtNameEmpty())
+                    binding.clUserName.setBackgroundResource(R.drawable.rectangle_border_gray_10)
+            }
+        }
+
+        binding.etId.setOnFocusChangeListener { _, hasFocus ->
+            if(hasFocus)
+                binding.clId.setBackgroundResource(R.drawable.rectangle_border_sky_10)
+            else {
+                if(isEtIdEmpty())
+                    binding.clId.setBackgroundResource(R.drawable.rectangle_border_gray_10)
+            }
+        }
+
+        binding.etPassword.setOnFocusChangeListener { _, hasFocus ->
+            if(hasFocus)
+                binding.clPassword.setBackgroundResource(R.drawable.rectangle_border_sky_10)
+            else {
+                if(isEtPassword())
+                    binding.clPassword.setBackgroundResource(R.drawable.rectangle_border_gray_10)
+            }
+        }
+    }
+
+    private fun initClickEvent() {
+        binding.btnSignUp.setOnClickListener {
+            if(isAllEditTextEmpty())
+                shortToast(getString(R.string.is_empty))
+            else {
+                val name = binding.etName.text.toString()
+                val id = binding.etId.text.toString()
+                val password = binding.etPassword.text.toString()
 
+                viewModel.insert(UserData(null, name, id, password))
+
+                setResult(Activity.RESULT_OK, Intent().putExtra("name", name)
+                    .putExtra("id", id)
+                    .putExtra("password", password)
+                )
+                shortToast(getString(R.string.sign_up_done))
+                finish()
+            }
+        }
+    }
+
+    private fun isAllEditTextEmpty() : Boolean {
+         return isEtNameEmpty() || isEtIdEmpty() || isEtPassword()
     }
```

Comment:
```suggestion
    private fun isAllEditTextEmpty() = isEtNameEmpty() || isEtIdEmpty() || isEtPassword()
```
이렇게 한 줄로도 작성 가능할 것 같습니다!!

Reply:
return 한 줄이면 `=` 써서 적는게 좀 더 깔끔하긴하죵 :) 좋은 리뷰 감사합니다 👍